Zooming out update (#6019)
DataDog/documentation: content/en/graphing/functions/rollup.md

`.rollup()`

*Recommended for expert users only.*

> Appending the `.rollup()` function at the end of a query allows you to do a custom [time aggregation][1], i.e. it defines what are the time intervals for your graph and how data points are aggregated within a given time interval.

Reviewer: ```suggestion
Appending the `.rollup()` function at the end of a query allows you to perform custom [time aggregation][1]. This function enables you to define the time intervals for your graph and how data points are aggregated within a given time interval.
```

> Appending the `.rollup()` function at the end of a query allows you to perform custom [time aggregation][1]. This function enables you to define the time intervals for your graph and how data points are aggregated within a given time interval.

Reviewer 2: Hm, can we add to the end of this?

> This function enables you to define the time intervals for your graph and how data points are aggregated within a given time interval.

Instead, can we just mention the limit? I'm afraid of someone referencing the doc and not reading the bottom section, this bit out of context would be puzzling if the rollup was ignored.

Could you get behind:

> This function enables you to define the time intervals for your graph (if larger than the query-enforced rollup interval) and how data points are aggregated within a given time interval.

Something to that effect? Just to call out that there IS a caveat.

> Appending the `.rollup()` function at the end of a query allows you to perform custom [time aggregation][1]. This function enables you to define the time intervals for your graph and how data points are aggregated within a given time interval.

Author: I changed the wording and layout.

The function takes two parameters, `<METHOD>` and `<TIME>`: `.rollup(<METHOD>,<TIME>)`.

| Parameter  | Description                                                                                                     |
|------------|-----------------------------------------------------------------------------------------------------------------|
| `<METHOD>` | Can be `sum`/`min`/`max`/`count`/`avg` and defines how data points are aggregated within a given time interval. |

> | `<TIME>`   | Time in second of the interval between two data points displayed.                                               |

Reviewer: ```suggestion
| `<TIME>`   | Time (in seconds) of the interval between two data points displayed.                                               |
```

> You can use either one individually, or both together like `.rollup(sum,120)` for instance. Find below bar graph displaying a week's worth of cpu usage for a host without using the `.rollup()` function:

Reviewer: ```suggestion
You can use them individually or together, for instance `.rollup(sum,120)`. The following bar graph displays a week's worth of CPU usage for a host **without** using the `.rollup()` function:
```

{{< img src="graphing/functions/rollup/smooth_1.png" alt="smooth_1" responsive="true" style="width:60%;" >}}

> And now it is the same metric, graphed using a day-long rollup with `.rollup(avg,86400)`:

Reviewer: ```suggestion
The following bar graph displays the same metric, graphed using a day-long rollup with `.rollup(avg,86400)`:
```

{{< img src="graphing/functions/rollup/smooth_2.png" alt="smooth_2" responsive="true" style="width:60%;" >}}

> ## Maximum number of points displayed

Author: I like the second one, i tweak it a bit to remove the "limit" wording.

When graphing, Datadog imposes a limit of 350 points per graph. In order to respect this limit, Datadog rolls up data points automatically with the `avg` method, effectively displaying the average of all data points within a time interval for a given metric.

> If a custom `.rollup()` function is applied and is lower than the Datadog limit, the Datadog limit is used instead. For example, if you're requesting `.rollup(20)` for a month-long window, data is returned at a rollup greater than 20 seconds in order to prevent returning more than 350 points.

Reviewer 2: > If a custom `.rollup()` function is applied and is lower than the Datadog limit, the Datadog limit is used instead. For example, if you're requesting `.rollup(20)` for a month-long window, data is returned at a rollup greater than 20 seconds in order to prevent returning more than 350 points.

Let's insert a line before this to clarify. Something to the effect of (this isn't the cleanest copy, just a suggestion):

"A custom `.rollup()` function can be used to set the type of time aggregation applied (avg, min, max, count, sum) and the time interval to rollup. However, if a custom `.rollup()` function is applied and uses a smaller interval than the Datadog limit, the Datadog limit is used instead, while still using the specified rollup method.

**Note**: Queries for `COUNT` and `RATE` type metrics have the `.as_count()` modifier appended automatically in the UI, which sets the rollup method used to `sum` and disables interpolation. This `.as_count()` is explicitly visible at the end of the query:

  {{< img src="graphing/functions/rollup/as_count.png" alt="as_count" responsive="true" style="width:50%;">}}

> For more details about how to use `.as_count()` and `.as_rate()` see [our blog post][2] or learn more about the effects of those functions with the [detailed documentation about in-application modifiers][3].

Reviewer: ```suggestion
For more details about how to use `.as_count()` and `.as_rate()` see the [blog post][2] or learn more about the effects of those functions with the [documentation on in-application modifiers][3].
```

## Rollups in monitors

> Rollups should usually be avoided in [monitor][4] queries, because of the possibility of misalignment between the rollup interval and the evaluation window of the monitor. The start and end of rollup intervals are aligned to UNIX time, not to the start and end of monitor queries, which means that a monitor may evaluate (and trigger on) an incomplete rollup interval containing only a small sample of data. To avoid this issue, delay the evaluation of your monitor by (at least) the length of the setup rollup interval.

Reviewer: ```suggestion
Rollups should usually be avoided in [monitor][4] queries, because of the possibility of misalignment between the rollup interval and the evaluation window of the monitor. The start and end of rollup intervals are aligned to UNIX time, not to the start and end of monitor queries. Therefore, a monitor may evaluate (and trigger on) an incomplete rollup interval containing only a small sample of data. To avoid this issue, delay the evaluation of your monitor by (at least) the length of the setup rollup interval.
```
